### PR TITLE
Enable 'no-unnecessary-type-assertion' rule

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: #0000
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Overview of change:

Just enables the rule and fixes new lint failures.
The two disables look like bugs an inconsistency between typescript's language service and compilation.

### CODE REVIEW

File: src/rules/noEmptyInterfaceRule.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ function walk(ctx: Lint.WalkContext<void>) {
             (node.heritageClauses === undefined ||
              node.heritageClauses[0].types === undefined ||
```
Comment (ajafff) [inline src/rules/noEmptyInterfaceRule.ts]: you can remove this line, because it can never be undefined

Comment (andy-hanson) [inline src/rules/noEmptyInterfaceRule.ts]: `strict-type-predicates` would have caught that. ⚡️

Comment (ajafff) [inline src/rules/noEmptyInterfaceRule.ts]: true, but this change of the type was only introduced recently with typescript@2.3

I'd like to enable `strict-type-predicates`, too. But there will be problems with some TypeChecker APIs that are not yet compatible with strictNullChecks

**Comment by adidahiya** [comment]
still some merge conflicts

**Pull request merged**